src/aspire/classification/rir_class2d.py

```diff
@@ -148,6 +113,54 @@ def __init__(
             )
         self._bispectrum = bispectrum_implementations[bispectrum_implementation]
 
+        # For now, only run with FSPCA basis
+        if pca_basis and not isinstance(pca_basis, FSPCABasis):
+            raise NotImplementedError(
+                "RIRClass2D has currently only been developed for pca_basis as a FSPCABasis."
+            )
+        self.pca_basis = pca_basis
+
+        # When a user provides a basis, the fspca_components arg is either
+        # redudant (same) or conflicting (different).
+        # So when a user provides fspca_components, we need to check it matches the basis' compoenents.
+        _provided_fspca_components = (
+            fspca_components is not self.__init__.__defaults__[1]
+        )
+        if pca_basis and _provided_fspca_components:
+            # Check the provided components match.
+            if pca_basis.components != fspca_components:
+                raise RuntimeError(
+                    f"`pca_basis` components {pca_basis.components} != {fspca_components} `fspca_components` provided by user."
+                )
+        elif pca_basis:  # fspca_components not specfied (default to taking from basis)
+            fspca_components = pca_basis.components
+
+            # For small problems, such as unit tests, we also need to guard against
+            #   requesting more fspca_components than exist in the basis now,
+            # In the case RIRClass2d instantiates the pca_basis,
+            #   this will be checked then, in FSPCABasis.
+            pca_basis._check_components()
+
```

Comment:
So the user can provide `pca_basis`, `fspca_components`, both, or neither, and the result would be:

`pca_basis` only: `fspca_components` is set to `pca_basis.count`
`fspca_components` only: `fspca_components` can be anything greater than `bispectrum_components`
Both: `fspca_components` must be equal to `pca_basis.components` but if the user inputs 400 it's set to `pca_basis.components` silently
Neither: `fspca_components = 400`

(Wrote it out mostly for myself to understand) Just wanted to confirm whether we still want the 400 default since we had gotten rid of it in `FSPCABasis`?

Reply:
Yeah I think this is good! Thanks